sdk/eventhub/azure-eventhub/samples/sync_samples/client_creation.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+"""
+Examples to show how to create EventHubProducerClient/EventHubConsumerClient.
+"""
+import os
+from azure.eventhub import (
+    EventHubProducerClient,
+    EventHubConsumerClient,
+    TransportType,
+    EventHubSharedKeyCredential
+)
+
+
+CONNECTION_STRING = os.environ['EVENT_HUB_CONN_STR']
+FULLY_QUALIFIED_NAMESPACE = os.environ['EVENT_HUB_HOSTNAME']
+EVENTHUB_NAME = os.environ['EVENT_HUB_NAME']
+SAS_POLICY = os.environ['EVENT_HUB_SAS_POLICY']
+SAS_KEY = os.environ['EVENT_HUB_SAS_KEY']
+CONSUMER_GROUP = "$Default"
+
+
+def create_producer_client():
+    print('Examples showing how to create producer client.')
+
+    # Create producer client from connection string.
+
+    producer_client = EventHubProducerClient.from_CONNECTION_STRING(
+        conn_str=CONNECTION_STRING  # connection string contains EventHub name.
+    )
+
+    # Illustration of commonly used parameters.
+    producer_client = EventHubProducerClient.from_CONNECTION_STRING(
+        conn_str=CONNECTION_STRING,
+        eventhub_name=EVENTHUB_NAME,  # EventHub name should be specified if it doesn't show up in connection string.
+        logging_enable=False,  # To enable network tracing log, set logging_enable to True.
+        retry_total=3,  # Retry up to 3 times to re-do failed operations.
```

Comment:
Add timeout for producer? It's tightly related to retry_total.

Reply:
timeout for producer is the` send_batch` method, not in the constructor.

Reply:
timeout for producer is the send_batch method, not in the constructor